OMCompiler/SimulationRuntime/c/simulation/solver/delay.c

```diff
@@ -41,138 +41,197 @@
+  /* Return expression value before simulation start */
+  if(time <= data->simulationInfo->startTime)
   {
-    infoStreamPrint(LOG_EVENTS_V, 0, "delayImpl: Entered at time < starting time: %g.", exprValue);
     return (exprValue);
   }
 
-  if(delayTime < 0.0)
-  {
-    throwStreamPrint(threadData, "Negative delay requested %g", delayTime);
-  }
-
+  /*  Empty delay buffer at initialization phase */
   if(length == 0)
   {
-    /*  This occurs in the initialization phase */
     infoStreamPrint(LOG_EVENTS, 0, "delayImpl: Missing initial value, using argument value %g instead.", exprValue);
     return (exprValue);
   }
 
-  /*
-   * Returns: expr(time?delayTime) for time>time.start + delayTime and
-   *          expr(time.start) for time <= time.start + delayTime.
-   * The arguments, i.e., expr, delayTime and delayMax, need to be subtypes of Real.
-   * DelayMax needs to be additionally a parameter expression.
-   * The following relation shall hold: 0 <= delayTime <= delayMax,
-   * otherwise an error occurs. If delayMax is not supplied in the argument list,
-   * delayTime need to be a parameter expression. See also Section 3.7.2.1.
-   * For non-scalar arguments the function is vectorized according to Section 10.6.12.
-   */
-  if(time <= data->simulationInfo->tStart + delayTime)
+  /* Return oldest element in ring buffer */
+  if(time <= data->simulationInfo->startTime + delayTime)
   {
-    double res = ((TIME_AND_VALUE*)getRingData(delayStruct, 0))->value;
-    infoStreamPrint(LOG_EVENTS_V, 0, "findTime: time <= tStart + delayTime: [%d] = %g",exprNumber, res);
-    return res;
+    return ((TIME_AND_VALUE*)getRingData(delayStruct, 0))->value;
   }
+  /* return expr(time-delayTime) */
   else
   {
-    /* return expr(time-delayTime) */
-    double timeStamp = time - delayTime;
-    double time0, time1, value0, value1;
-    int i;
-
-    assertStreamPrint(threadData, 0.0 <= delayTime, "Negative delay requested: delayTime = %g", delayTime);
+    timeStamp = time - delayTime;
 
     /* find the row for the lower limit */
     if(timeStamp > ((TIME_AND_VALUE*)getRingData(delayStruct, length - 1))->t)
     {
-      infoStreamPrint(LOG_EVENTS_V, 0, "delayImpl: find the row  %g = %g", timeStamp, ((TIME_AND_VALUE*)getRingData(delayStruct, length - 1))->t);
       /* delay between the last accepted time step and the current time */
       time0 = ((TIME_AND_VALUE*)getRingData(delayStruct, length - 1))->t;
       value0 = ((TIME_AND_VALUE*)getRingData(delayStruct, length - 1))->value;
       time1 = time;
       value1 = exprValue;
-      infoStreamPrint(LOG_EVENTS_V, 0, "delayImpl: times %g and %g", time0, time1);
-      infoStreamPrint(LOG_EVENTS_V, 0, "delayImpl: values %g and  %g", value0, value1);
     }
     else
     {
-      i = findTime(timeStamp, delayStruct);
+      i = findTime(timeStamp, delayStruct, &foundEvent);
+      if (foundEvent) {
+        // TODO: Do something if an event was found!
```

Comment:
Is this finished or did you forget this? :)

Reply:
Not really sure. I guess in that case I should just remove it.